YoutubeExplode/Utils/Extensions/HttpExtensions.cs

```diff
@@ -1,39 +1,19 @@
-﻿using System.IO;
-using System.Net;
-using System.Net.Http;
+﻿using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace YoutubeExplode.Utils.Extensions;
 
 internal static class HttpExtensions
 {
-    private class NonDisposableHttpContent : HttpContent
-    {
-        private readonly HttpContent _content;
-
-        public NonDisposableHttpContent(HttpContent content) => _content = content;
-
-        protected override async Task SerializeToStreamAsync(
-            Stream stream,
-            TransportContext? context
-        ) => await _content.CopyToAsync(stream);
-
-        protected override bool TryComputeLength(out long length)
-        {
-            length = default;
-            return false;
-        }
-    }
-
-    public static HttpRequestMessage Clone(this HttpRequestMessage request)
+    public static async ValueTask<HttpRequestMessage> CloneAsync(this HttpRequestMessage request)
     {
         var clonedRequest = new HttpRequestMessage(request.Method, request.RequestUri)
         {
             Version = request.Version,
             // Don't dispose the original request's content
             Content = request.Content is not null
-                ? new NonDisposableHttpContent(request.Content)
```

Comment:
Would it make more sense to use `new StreamHttpContent(await request.Content.ReadAsStreamAsync())` as a more general solution?

Reply:
I just tested with `new StreamContent(await request.Content.ReadAsStreamAsync())` and it doesn't work unfortunately. It would return an `Http 400 Bad Request` again.

Reply:
What about `new ByteArrayContent(await request.Content.ReadAsBytesAsync())`?

Reply:
I tried it and still returns the same error. I will look into detail why using a `ByteArrayContent` or `StreamContent` doesn't work on Android.

Reply:
Yes, please do. It's really weird that only `StringContent` works. Ideally, we should re-encoding the data from/to string because it may be wasteful.

Reply:
I just did some more testing and noticed that initializing a new `StringContent` has default headers for posting strings, such as `Content-Type` along with `CharSet` headers. Using `new StreamContent(...)` doesn't copy headers from `StringContent` so I updated my pull request to copy the headers from the `HttpContent`. I tested it and it works on mobile devices. Without the headers, it throws an exception but not on Desktop.

Reply:
Interesting! Thanks